File: lib/domain/entities/user_model.dart
class UserModel {
  final String id;
  final String username;
  final DateTime createdAt;

  UserModel({
    required this.id,
    required this.username,
    required this.createdAt,
  });

  UserModel copyWith({String? id, String? username, DateTime? createdAt}) {
    return UserModel(
      id: id ?? this.id,
      username: username ?? this.username,
      createdAt: createdAt ?? this.createdAt,
    );
  }

  @override
  String toString() {
    return 'UserModel{id: $id, username: $username, createdAt: $createdAt}';
  }
}

File: server/models/UserModel.js
const { v4: uuidv4 } = require('uuid');

class UserModel {
    constructor(username) {
        this.id = uuidv4();
        this.username = username;
        this.createdAt = new Date();
        this.isGuest = true;
    }
}

module.exports = UserModel;
